lib/ansible/module_utils/facts/system/pkg_mgr.py

```diff
@@ -78,10 +92,21 @@ def collect(self, module=None, collected_facts=None):
             if os.path.exists(pkg['path']):
                 pkg_mgr_name = pkg['name']
 
-        if pkg_mgr_name == 'apt' and \
-                os.path.exists('/usr/bin/rpm') and \
-                not os.path.exists('/usr/bin/dpkg'):
-            pkg_mgr_name = 'apt_rpm'
+        # apt is easily installable and supported by distros other than those
+        # that are debian based, this handles some of those scenarios as they
+        # are reported/requested
+        if pkg_mgr_name == 'apt' and collected_facts['ansible_os_family'] in ["RedHat", "Altlinux"]:
+            if collected_facts['ansible_os_family'] == 'RedHat':
+                pkg_mgr_name = self._check_rh_versions(collected_facts)
+
+            elif collected_facts['ansible_os_family'] == 'Altlinux':
+                pkg_mgr_name = 'apt_rpm'
+
+        # pacman has become available by distros other than those that are Arch
+        # based by virtue of a dependency to the systemd mkosi project, this
+        # handles some of those scenarios as they are reported/requested
+        if pkg_mgr_name == 'pacman' and collected_facts['ansible_os_family'] in ["RedHat"]:
+            pkg_mgr_name = self._check_rh_versions(collected_facts)
```

Comment:
Maybe organizing this with one conditional per ansible_os_family will be better for the future (and allow handling a distro being able to install more package managers in the future)

```
if collected_facts['ansible_os_family'] == "RedHat":
    if pkg_mgr_name not in ('yum', 'dnf'):
         pkg_mgr_name = self._check_rh_versions(collected_facts) 
elif collected_facts['ansible_os_family'] == 'Altlinux':
        pkg_mgr_name = 'apt_rpm'
```